server/pxf-api/src/main/java/org/greenplum/pxf/api/utilities/FragmenterCacheFactory.java

```diff
@@ -0,0 +1,44 @@
+package org.greenplum.pxf.api.utilities;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.RemovalListener;
+import org.greenplum.pxf.api.model.Fragment;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Factory class for creation of {@link com.google.common.cache.Cache} objects.
+ */
+public class FragmenterCacheFactory {
+
+    private final Logger LOG = LoggerFactory.getLogger(this.getClass());
+
+    /**
+     * Singleton instance of the FragmenterCacheFactory
+     */
+    private static final FragmenterCacheFactory instance = new FragmenterCacheFactory();
+
+    private final Cache<String, List<Fragment>> fragmenterCache = CacheBuilder.newBuilder()
+            .expireAfterAccess(10, TimeUnit.SECONDS)
+            .removalListener((RemovalListener<String, List<Fragment>>) notification ->
+                    LOG.debug("Remove fragmenterCache entry for transactionId {}", notification.getKey()))
```

Comment:
If this happens after the removal has occurred, maybe change to past tense "Removed.." and list how many fragments were cached.
```suggestion
                    LOG.debug("Removed fragmenterCache entry for transactionId {} with {} fragments", notification.getKey(), notification.getValue().size()))
```
Or even better yet to use extended API to check for the reason: https://google.github.io/guava/releases/23.0/api/docs/com/google/common/cache/RemovalNotification.html

Also, not clear from the refs above whether the values / keys can be nulls, in which case, finding size would need to be guarded by checking for nulls.